### ISSUE TITLE
GSoD: Improving the contributors guide

### ISSUE THREAD

**Issue by nanjekyejoannah** [geomstats/geomstats#1574]
Moving this to an issue for public feedback @lpereira95 @ninamiolane 

# Introduction

The contributing guide is the face documentation for any open source project contributors. The goal for the re-organization is to:

- Add any missing sections making it holistic
- Elaborate existing sections with more relevant contribution detail

This outline proposes the main sections of the new contributors guide.

## Proposed Documentation Structure

We are using the Diátaxis documentation structure. The structure dictates that the documentation is divided into four main parts; *Getting started, How-to guides, Reference, and Explanation*. The structure gives a clear scope and sets a clear tone on where each piece of information should go,  for both users and documentation authors. 

The contributor guide is currently a separate section outside these major categories. In future it may make sense to put it under the *Getting Started* section. The main sections for the contribution guide are listed below:

- Getting started for developers
    - Installation
    - Using Geomstats
        - Instantiate a manifold
        - Instantiate an algorithm
        -  Run the algorithm
- Development Setup
    - Source control with Git
    - Getting the source code
        - Forking the repository
        - Pulling the code
    - Dependencies
    - Choosing a Backend
    - Run the tests
    - Build the docs
    - Debugging
- Folder structure
    - geomstats
    - docs
    - examples
    - Notebooks
    - tests
- Testing
    - What is TDD
    - TDD with pytest
    - Writing tests for Geomstats
    - Running tests
        - The whole test suite
        - A single test file
        - A test case
    - Running the benchmarks
- Documentation
    - Introduction to RST
    - Categories
    - Guidelines and best practices
    - Testing
    - Docstrings style guide
- Coding style
    - PEP 8
    - code style guidelines i.e., classes, methods, variable etc
    - Black formatter
- Workflow of a contribution
    - Create or choose an issue
    - Create a new branch
    - Making changes
    - Run tests
    - commit and push
    - Review
    - Merge
    - Pull request checklist
    - Deleting merged branch
- Reporting bugs and features
    - Issue tracker
    - Template of a bug/issue report
- Issue Triaging
    - Reproducing issues
    - Commenting on alternative solutions
    - Answering questions
    - Labelling and assigning the issue

**Comment by ninamiolane** [comment]
Thanks @nanjekyejoannah and @lpereira95 ! 

Minor comments: 
- I think the "Getting started for developers" could actually be just a "Getting started". The workflow "Instantiate a manifold, instantiate an algorithm, run the algorithms", is the "Getting started" for users too.
- Any chance the code corresponding to the workflow "Instantiate a manifold, instantiate an algorithm, run the algorithms" could be run within the doc? I always fear that this "getting started" code becomes outdated.

**Comment by lpereira95** [comment]
@ninamiolane, I think [`sphinx.ext.doctest`](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html) may answer the second point.

**Comment by nanjekyejoannah** [comment]
@ninamiolane yes, that is possible. Also, let us have consensus, do we include the `Get started` section?
Seems @lpereira95 was not very enthusiastic while @ninamiolane seems not to object.

**Comment by ninamiolane** [comment]
I agree with Luis that the (potentially long section on) "Getting started for developers" is different than a smaller "Getting started". 

But what is currently in the section currently called "Getting started for developers" is not specific to developers, thus this section can be renamed "Getting started"?

**Comment by lpereira95** [comment]
What would such a section contain? In this outline it looks exactly the same as the already existing "Getting started". That's why I don't think it should be added. Meanwhile, if we find useful content to be added to a Contributing Guide "Getting started", I do not oppose. We should just remember that a developer has less "basic" needs (i.e. is more independent) than a user.

**Comment by nanjekyejoannah** [comment]
For now, we will leave out that section for now.
